api/views/default.py

```diff
@@ -911,6 +913,13 @@ def create_submit(request: django.http.HttpRequest, task_assignment: int) -> Jso
             },
             status=400,
         )
+    except SubmitPastHardDeadline:
+        return JsonResponse(
+            {
+                "error": "The submit was send after a deadline, that was set as final, so it is not valid submit"
```

Comment:
Please, reformulate the error message to: The submission was sent after the deadline, which was final, so it is not a valid submission.

Including the trailing "dot" :-)